Sources/Sentry/Transactions/SentrySpanContext.m

```diff
@@ -0,0 +1,59 @@
+//
+//  SpanContext.m
+//  Sentry
+//
+//  Created by Dhiogo Brustolin on 05/01/21.
+//  Copyright © 2021 Sentry. All rights reserved.
+//
+
+#import "SentrySpanContext.h"
+#import "SentryId.h"
+#import "SentrySpanId.h"
+
+@interface
+SentrySpanContext () {
+    NSMutableDictionary<NSString *, NSString *> *_tags;
+}
+
+@end
+
+@implementation SentrySpanContext
+
+- (instancetype)init
+{
+    return [self initWithtraceId:[[SentryId alloc] init]
+                          spanId:[[SentrySpanId alloc] init]
+                        parentId:nil
+                      andSampled:false];
+}
+
+- (instancetype)initWithSampled:(BOOL)sampled
+{
+    return [self initWithtraceId:[[SentryId alloc] init]
+                          spanId:[[SentrySpanId alloc] init]
+                        parentId:nil
+                      andSampled:sampled];
+}
+
+- (instancetype)initWithtraceId:(SentryId *)traceId
+                         spanId:(SentrySpanId *)spanID
+                       parentId:(SentrySpanId *_Nullable)parentId
+                     andSampled:(BOOL)sampled
+{
+    if (self = [super init]) {
+        self.traceId = traceId;
+        self.spanId = spanID;
+        self.parentSpanId = parentId;
+        self.sampled = sampled;
+    }
+    return self;
+}
+
+- (NSMutableDictionary *)tags
+{
+    if (_tags == nil)
+        _tags = [[NSMutableDictionary alloc] init];
```

Comment:
`l`: Why not initialize the dict in the constructor so we can remove the check for `nil` in here.

Reply:
This part was copied from SentryEvent. I guess we could.

Reply:
Initializing in the constructor only we're certain this will always be used otherwise we're just allocating for no use.

Reply:
I didn't think about that @bruno-garcia, so I'm fine with keeping it then.